### DISCUSSION TITLE
Upload node-generated file to Strapi

### DISCUSSION THREAD

**Discussion by landorid** [strapi/strapi#8110] [Legacy Topics]
I have a pdf created on my node server. I want to link and upload it to a content type. The pdf is generated during the `create` method. How should I pass the stream or buffer to the service to upload the file?

```js
const pdf = await createPDF(ctx.request.body, new Date());
    
let entity = await strapi.services.questionnaire.create(ctx.request.body, {
    files: {
      report: {
        path: pdf, // error:  The argument 'path' must be a string or Uint8Array without null bytes.
      },
    },
});
``` 

Thank you!

**Comment by derrickmehaffy** [comment]
Hi @landorid I'm reaching out to our backend engineers to see if they have a good suggestion as I've seen a few questions like this come up.
